.github/workflows/generate-summary.yaml

```diff
@@ -564,15 +579,34 @@ jobs:
       # These three conditions should be equivalent, however, add them just in case
       # we change something accidentally which breaks it
       - name: Add Valid-Baseline Label to New Issue
-        uses: actions/github-script@v6
         if: ${{ github.workflow == 'Run-Frequent' && inputs.prefix == '' && inputs.issue_hash_prefix == 'Testsuite Status' && steps.issue-labels.outputs.valid_baseline != 'false' }}
+        uses: ./.github/actions/common/save-label
+        with:
+          label: "'valid-baseline'"
+          output: ${{ steps.label-aggregation-preset.outputs.label_file }}
+
+      # Execute this step even if there was a previous failure. Labels must always be uploaded to the issue
+      - name: Labels upload setup
+        id: labels-upload-setup
+        if: ${{ success() || failure() }}
```

Comment:
nit: can replace this with always()

Reply:
always() runs even if the run is cancelled, and wanted to skip the label creation for cancelled runs.

Reply:
Then we should be able to use !cancelled() as the condition. Is there a situation where `success() == false || failure() == true`?

Reply:
Yes. If any of the previous steps fails, but the label do exist for the current run, `success() == false, failure() == true` occurs. If I only use `if: ${{ failure() }}` this step won't execute if the previous steps succeeded

Reply:
To make the conditional easier to understand, if we just want to ensure it always runs without cancellation, we should use !cancelled()
https://github.com/ewlu/action_testing/actions/runs/10817079111/job/30009682652